pkg/services/accesscontrol/middleware.go

```diff
@@ -65,7 +67,7 @@ func Deny(c *models.ReqContext, evaluator accesscontrol.Evaluator, err error) {
 	// internal server error or access denied.
 	c.JSON(http.StatusForbidden, map[string]string{
 		"title":         "Access denied", // the component needs to pick this up
-		"message":       fmt.Sprintf("You'll need additional permissions to perform this action. Permissions needed: %s", evaluator.String()),
+		"message":       fmt.Sprintf("You'll need additional permissions to perform this action. Permissions needed: %s", permissions),
```

Comment:
This is a significant change.

```
func (p permissionEvaluator) String() string {
	return p.Action
}

func (p permissionEvaluator) GoString() string {
	return fmt.Sprintf("action:%s scopes:%s", p.Action, strings.Join(p.Scopes, ", "))
}
```

Reply:
Yes you are  right, i will revert that one